pan3d/dataset_builder.py

```diff
@@ -261,77 +281,74 @@ def on_set_array_active(self, array_active, **kwargs):
                     "step": 1,
                 }
             )
-            self.state.expanded_coordinates.append(key),
+            self.state.ui_expanded_coordinates.append(key),
         self.auto_select_coordinates()
-        self.state.dirty("coordinates", "expanded_coordinates")
+        self.state.dirty("da_coordinates", "ui_expanded_coordinates")
+        self.plotter.clear()
+        self.plotter.view_isometric()
 
-    @change("x_array")
-    def on_set_x_array(self, x_array, **kwargs):
-        self.algorithm.x = x_array
+    @change("da_x")
+    def on_set_da_x(self, da_x, **kwargs):
+        self.algorithm.x = da_x
         self.mesh_changed()
 
-    @change("y_array")
-    def on_set_y_array(self, y_array, **kwargs):
-        self.algorithm.y = y_array
+    @change("da_y")
+    def on_set_da_y(self, da_y, **kwargs):
+        self.algorithm.y = da_y
         self.mesh_changed()
 
-    @change("z_array")
-    def on_set_z_array(self, z_array, **kwargs):
-        self.algorithm.z = z_array
+    @change("da_z")
+    def on_set_da_z(self, da_z, **kwargs):
+        self.algorithm.z = da_z
         self.mesh_changed()
 
-    @change("t_array")
-    def on_set_t_array(self, t_array, **kwargs):
-        self.algorithm.time = t_array
-        if self.dataset and self.state.array_active and t_array:
-            time_steps = self.dataset[self.state.array_active][t_array]
+    @change("da_t")
+    def on_set_da_t(self, da_t, **kwargs):
+        self.algorithm.time = da_t
+        if self.dataset and self.state.da_active and da_t:
+            time_steps = self.dataset[self.state.da_active][da_t]
             # Set the time_max in the state for the slider
-            self.state.t_max = len(time_steps) - 1
+            self.state.da_t_max = len(time_steps) - 1
             self.mesh_changed()
 
-    @change("t_index")
-    def on_set_t_index(self, t_index, **kwargs):
-        self.algorithm.time_index = int(t_index)
+    @change("da_t_index")
+    def on_set_da_t_index(self, da_t_index, **kwargs):
+        self.algorithm.time_index = int(da_t_index)
         self.mesh_changed()
 
-    @change("view_edge_visibility")
-    def on_set_edge_visiblity(self, view_edge_visibility, **kwargs):
-        if self.actor is None:
-            return
-        self.actor.GetProperty().SetEdgeVisibility(1 if view_edge_visibility else 0)
-        self.plotter.update()
-
-    @change("x_scale", "y_scale", "z_scale")
-    def on_set_scale(self, x_scale=None, y_scale=None, z_scale=None, **kwargs):
-        x_scale = x_scale or self.state.x_scale
-        y_scale = y_scale or self.state.y_scale
-        z_scale = z_scale or self.state.z_scale
-        self.plotter.set_scale(xscale=x_scale, yscale=y_scale, zscale=z_scale)
+    @change("ui_dialog_shown")
+    def on_set_ui_dialog_shown(self, ui_dialog_shown, **kwargs):
```

Comment:
`on_dialog_visibility_change(...`

Reply:
I'm hoping to have a consistent naming convention for these state change functions. Right now it's `on_set_[name]`, but if you think it should be `on_[name]_change` I can switch it for all of them, but I do think the function name should include the name of the state variable somewhere.

Reply:
Agree it should be consistent and be like `on_[xxx]_change`, but having the variable name in both the annotation and method name does not seem that useful. Having the intent conveyed by the method name is better. See the grouping comment `on_coordinates_change`

Reply:
I am hesitant to lose the congruence between the annotation and method name. To me it feels messy to refer to the same property by different names (`ui_dialog_shown` and `dialog_visibility`). If the nomenclature of "visibility" is preferred over "shown", I would change it everywhere, not just in the function name.

Reply:
I technically agree, especially if it is a 1-to-1 mapping. But it is still not a good name yet.

Looking at the code, I just realized it is not a boolean but more the name of the dialog you want to show.

What about ui_mode/action? Then a method like `on_mode_change` should be appropriate. No need to repeat the ui_* prefix in the method name. 

Your naming should convey the intent not how you do it (i.e. with a dialog).